src/features/dataset/DatasetHeader.tsx

```diff
@@ -45,14 +65,42 @@ const DatasetHeader: React.FC<DatasetHeaderProps> = ({
       })
   }
 
+  const handleTitleChange = (title: string, commitTitle: string) => {
+    commitDatasetTitle(qriRef, title, commitTitle)(dispatch)
+      .then(({ type }) => {
+        if (type === 'API_TITLE_SUCCESS') {
+          const ref = newQriRef({ username: qriRef.username, name: qriRef.name })
+          loadDataset(ref)(dispatch).then(({ type }) => {
+            if (type === 'API_DATASET_SUCCESS') {
+              dispatch(clearModal())
+            }
+          })
+          dispatch(loadDatasetCommits(ref))
+        }
+      })
+  }
+
+  const onTitleClick = () => {
+    if (!isNew) {
+      const left = location.pathname === `/${header.username}/${header.name}` ? '-206px' : '153px'
+      const top = location.pathname === `/${header.username}/${header.name}` ? '67px' : '65px'
+      const position = location.pathname === `/${header.username}/${header.name}` ? 'relative' : 'absolute'
```

Comment:
so this is an interesting way of solving the location problem, but I'm worried that if we manipulate the page in the future, these numbers will be out of date.

Have you looked into using a callback to measure a DOM element: https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node

Reply:
+1